source/hpack.c

```diff
@@ -668,7 +685,20 @@ int aws_hpack_insert_header(struct aws_hpack_context *context, const struct aws_
 
     /* Put the header at the "front" of the table */
     struct aws_http_header *table_header = s_dynamic_table_get(context, 0);
-    *table_header = *header;
+
+    /* TODO:: We can optimize this with ring buffer. */
+    /* allocate memory for the name and value, which will be deallocated whenever the entry is evicted from the table or
+     * the table is cleaned up. We keep the pointer in the name pointer of each entry */
+    uint8_t *memory_buffer = aws_mem_calloc(context->allocator, header->name.len + header->value.len, sizeof(uint8_t));
```

Comment:
check that memory_buffer isn't NULL

Reply:
In general this could all be more unified using the aws_Byte_buf_append apis.

Reply:
I was trying to keep the size of the total buffer we allocate smaller than the max_table_size setting. So, we need to free the buffer when the entry is evicted. If we use byte_buf_append, we may need a "big enough" buffer, which may not meet the setting.